styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -102,64 +109,67 @@ public WorkflowResource(Storage storage, String schedulerServiceBaseUrl,
             rc -> patchState(arg("cid", rc), arg("wfid", rc), rc.request()))
     );
 
-    final List<Route<AsyncHandler<Response<ByteString>>>> forwardedRoutes = Arrays.asList(
-        Route.async(
-            "POST", BASE + "/<cid>",
-            rc -> createOrUpdateWorkflow(arg("cid", rc), rc)
-        ),
-        Route.async(
-            "DELETE", BASE + "/<cid>/<wfid>",
-            rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc), rc)
-        )
-    );
-
-    return cat(
-        Api.prefixRoutes(routes, V3),
-        Api.prefixRoutes(forwardedRoutes, V3)
-    );
+    return Api.prefixRoutes(routes, V3);
   }
 
-  private CompletionStage<Response<ByteString>> deleteWorkflow(String cid, String wfid,
-                                                               RequestContext rc) {
-    // TODO: handle workflow crud directly in api service instead of proxying to scheduler
-    return forwardingClient.send(rc.request().withUri(schedulerApiUrl("workflows", cid, wfid)));
+  private Response<ByteString> deleteWorkflow(String cid, String wfid) {
+    final WorkflowId workflowId = WorkflowId.create(cid, wfid);
+    try {
+      storage.delete(workflowId);
+    } catch (IOException e) {
+      final String message = String.format("Couldn't remove workflow %s. ", workflowId);
```

Comment:
Ideally this would return `404` if the workflow didn't exist but datastore delete doesn't tell...

Reply:
The `SchedulerResource` checked if the workflow existed before attempting to delete it and returned `404` if it didn't exist. Let's do that here as well.

Reply:
That was actually a strange design to me. But yes I guess we need to keep backward compatibility.

Reply:
Returning `404` if something that was expected to exists didn't exist makes sense to me.